plugins/web-terminal/src/components/TerminalComponent/TerminalComponent.tsx

```diff
@@ -1,27 +1,108 @@
-import { makeStyles } from '@material-ui/core';
-import React from 'react';
+import { Button, makeStyles, TextField } from '@material-ui/core';
+import React, { useCallback } from 'react';
 import { Terminal } from 'xterm';
 import { FitAddon } from 'xterm-addon-fit';
+import { AttachAddon } from 'xterm-addon-attach';
 import './static/xterm.css';
+import { InfoCard, Progress } from '@backstage/core-components';
 
 const useTerminalStyles = makeStyles({
   term: {
     width: '100%',
-    height: '100%',
+    height: '400px',
+  },
+  formDisplay: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    flexDirection: 'column',
+    rowGap: '10px',
+    margin: '10px',
+  },
+  centerInfo: {
+    textAlign: 'center',
+    marginTop: '10px',
+    marginBottom: '10px',
   },
 });
+
 export const TerminalComponent = () => {
-  const fitAddon = new FitAddon();
-  const terminal = new Terminal();
-  terminal.loadAddon(fitAddon);
-  fitAddon.fit();
+  const [websocketRunning, setWebsocketRunning] = React.useState(false);
+  const [loading, setLoading] = React.useState(false);
+  const [token, setToken] = React.useState('');
+  const [cluster, setCluster] = React.useState('');
   const classes = useTerminalStyles();
+  const tokenRef = React.useRef<HTMLInputElement>(null);
+  const clusterRef = React.useRef<HTMLInputElement>(null);
+
+  /**
+   * Terminal is attached to created websocket, attachment allows user
+   * to interact with terminal as they would with a normal terminal
+   */
+  const setupTerminal = useCallback((ws: WebSocket) => {
+    const terminal = new Terminal();
+    const attachAddon = new AttachAddon(ws);
+    terminal.loadAddon(attachAddon);
+    const fitAddon = new FitAddon();
+    terminal.loadAddon(fitAddon);
+    fitAddon.fit();
+    const element = document.getElementById('terminal');
+    if (element !== null) {
+      terminal.open(element);
+    }
+  }, []);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (tokenRef.current) {
+      setToken(tokenRef.current.value);
+    }
+    if (clusterRef.current) {
+      setCluster(clusterRef.current.value);
+    }
+  };
+
   React.useEffect(() => {
-    const terminalElement = document.getElementById('terminal');
-    if (terminalElement) {
-      terminal.open(terminalElement);
-      terminal.write('$ ');
+    if (!token) {
+      return;
     }
-  });
-  return <div className={classes.term} id="terminal" />;
+    setLoading(true);
+    const ws = new WebSocket('ws://localhost:4000', [
```

Comment:
This won't work in deployment, since `localhost` is the client machine, not the server.

Reply:
Currently we don't have any websocketserver that could be this pointed to so this is supposed to be just a placeholder. I have reliazed that this is not a best design so I have added option to configure this. Example of the configuration is [here](https://github.com/operate-first/service-catalog/pull/193/commits/51cefbcf41f4d5d5bda9192067cdf42853020134#diff-cc2f0c480e34bbf87b5cffe388116011cc6953ea75ea96a6d641134a02fb99f7)